signer/tests/integration/setup.rs

```diff
@@ -898,7 +929,22 @@ impl TestSweepSetup2 {
             unsigned.tx.compute_txid()
         };
 
-        // Let's sweep in the transaction
+        let block_header = rpc.get_blockchain_info().unwrap();
+
+        self.broadcast_info = Some(BroadcastSweepTxInfo {
+            block_hash: block_header.best_block_hash,
+            txid,
+        });
+    }
+
+    /// This function generates a sweep transaction that sweeps in the
+    /// deposited funds and sweeps out the withdrawal funds in a proper
+    /// sweep transaction, that is also confirmed on bitcoin.
+    pub fn submit_sweep_tx(&mut self, rpc: &Client, faucet: &Faucet) {
+        self.broadcast_sweep_tx(rpc);
```

Comment:
should we skip `broadcast_sweep_tx` if `self.broadcast_info.is_some()`?

Reply:
Oh, yeah we should.

Reply:
I'll make the suggested change here, but later I think we should break this out as `broadcast_sweep_tx` and `confirm_sweep_tx`, where `confirm_sweep_tx` does not broadcast anything.

Reply:
Done in https://github.com/stacks-network/sbtc/pull/1411/commits/73f04b6ccaff811d15ca6ca82b7499f457b53797.